src/transformers/models/encoder_decoder/modeling_flax_encoder_decoder.py

```diff
@@ -0,0 +1,853 @@
+            outputs, past = outputs
+            outputs = outputs[:1] + (unfreeze(past["cache"]),) + outputs[1:]
+
+        return outputs
+
+    @add_start_docstrings_to_model_forward(ENCODER_DECODER_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=FlaxSeq2SeqLMOutput, config_class=_CONFIG_FOR_DOC)
+    def __call__(
+        self,
+        input_ids: jnp.ndarray,
+        attention_mask: Optional[jnp.ndarray] = None,
+        decoder_input_ids: Optional[jnp.ndarray] = None,
+        decoder_attention_mask: Optional[jnp.ndarray] = None,
+        position_ids: Optional[jnp.ndarray] = None,
+        decoder_position_ids: Optional[jnp.ndarray] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+        train: bool = False,
+        params: dict = None,
+        dropout_rng: PRNGKey = None,
+    ):
+        r"""
+        Returns:
+
+        Examples::
+
+            >>> from transformers import FlaxEncoderDecoderModel, BertTokenizer, GPT2Tokenizer
+
+            >>> bert_tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')
+            >>> gpt2_tokenizer = GPT2Tokenizer.from_pretrained('gpt2')
+            >>> model = FlaxEncoderDecoderModel.from_encoder_decoder_pretrained('bert-base-uncased', 'gpt2') # initialize Bert2GPT2 from pre-trained checkpoints
+
+            >>> # forward
+            >>> input_ids = bert_tokenizer.encode("Hello, my dog is cute", add_special_tokens=True, return_tensors='jax')  # Batch size 1
+            >>> decoder_input_ids = gpt2_tokenizer.encode("Hello, my dog is cute", add_special_tokens=True, return_tensors='jax')  # Batch size 1
+            >>> outputs = model(input_ids=input_ids, decoder_input_ids=input_ids)
+
+            >>> # save and load from pretrained
+            >>> model.save_pretrained("bert2gpt2")
+            >>> model = FlaxEncoderDecoderModel.from_pretrained("bert2gpt2")
+
+            >>> # generation (use GPT2's `bos_token_id` to start the generation.)
+            >>> generated = model.generate(input_ids, decoder_start_token_id=model.config.decoder.bos_token_id)
+
+        """
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        return_dict = return_dict if return_dict is not None else self.config.return_dict
+
+        # prepare encoder inputs
+        if attention_mask is None:
+            attention_mask = jnp.ones_like(input_ids)
+        if position_ids is None:
+            batch_size, sequence_length = input_ids.shape
+            position_ids = jnp.broadcast_to(jnp.arange(sequence_length)[None, :], (batch_size, sequence_length))
+
+        # prepare decoder inputs
+        if decoder_input_ids is None:
```

Comment:
This we can actually completely remove. The `decoder_input_ids` normally won't be generated from the `inptu_ids`. This is something which is very peculiar to Bart and MBart and not at all the default case => let's remove it and force the user to put in `decoder_input_ids`